### PR TITLE
fix: issue 253 resolved

### PR DESCRIPTION
**Pull request by amanueltsfy**
## Issues

image

### CODE REVIEW

File: focus-blocked/js/main.js [hunk 1]
```diff
@@ -147,7 +147,7 @@ try {
     }
   });
 
-  if (cuddly_bear_mode && !strict_blocking) {
+  if (cuddly_bear_mode && !strict_blocking && shouldAlwaysBlockFocusMode) {
```
[the remark this answers is not in the document]
Comment (jeznag) [reply focus-blocked/js/main.js:150]: @amanueltsfy isn't it going to not show the button if the URL was blocked by AI? I don't think the shouldAlwaysBlockFocusMode condition should be there

[the remark this answers is not in the document]
Comment (amanueltsfy) [reply focus-blocked/js/main.js:150]: we need it because you've mentioned the following issue
image

[the remark this answers is not in the document]
Comment (amanueltsfy) [reply focus-blocked/js/main.js]: I just updated the condition
image

File: focus-blocked/js/main.js [hunk 2]
```diff
@@ -147,7 +147,11 @@ try {
     }
   });
 
-  if (cuddly_bear_mode && !strict_blocking) {
+  if (
+    (cuddly_bear_mode &&
+      (shouldAlwaysBlockFocusMode || block_type === grizzly)) ||
+    (!strict_blocking && cuddly_bear_mode)
+  ) {
```
Comment (jeznag) [inline focus-blocked/js/main.js:154]: @amanueltsfy I'm still not sure this is handling the conditions we need to handle.

Here's the scenarios you need to test:

(in all cases, the user is in Cuddly Bear mode)
1. the URL was blocked by AI blocking - show button
2. the URL was a super distracting site and the user was not in a focus session - show button
3. the URL was a super distracting site that is allowed in the focus mode and the user was in a focus session in non strict blocking - show button
4. the URL was a super distracting site that is allowed in the focus mode and the user was in a focus session in  strict blocking - do not show button
5. the URL was a super distracting site that is NOT allowed in the focus mode and the user was in a focus session - do not show button

Please test those scenarios and add a loom video

I have a feeling that the original condition would be sufficient but let's see when you test it

Comment (amanueltsfy) [reply focus-blocked/js/main.js]: https://www.loom.com/share/cceba2b68b80439aba1024e3a27561dc?sid=0aa900f1-3b71-4071-b8e2-1e48a0858f2d

File: focus-blocked/js/main.js [hunk 3]
```diff
@@ -147,7 +147,13 @@ try {
     }
   });
 
-  if (cuddly_bear_mode && !strict_blocking) {
+  let focusRemainingSeconds = focus_mode_end_time.diff(moment(), 'seconds');
+  if (
+    (cuddly_bear_mode ||
+      shouldActivateAlwaysBlock ||
+      focusRemainingSeconds <= 0) &&
```
Comment (jeznag) [inline focus-blocked/js/main.js:154]: If remaining seconds is less than zero, the block is over and it should show unblocked?

**Review by jeznag — APPROVED** [review]

**Pull request merged**